ops/main.py

```diff
@@ -361,6 +361,15 @@ def _should_use_controller_storage(db_path: Path, meta: CharmMeta) -> bool:
         return False
 
 
+def _get_breaking_relation_id() -> Optional[int]:
+    """Returns the relation id of the currently breaking relation, if applicable."""
+    if not os.environ.get("JUJU_HOOK_NAME", "").endswith("-relation-broken"):
```

Comment:
We don't actually use `JUJU_HOOK_NAME` in ops anymore, we use the newer `JUJU_DISPATCH_PATH`. However, that's already parsed by the `_Dispatcher` class here, and the event name (with `-` already replaced with `_`) is available as `dispatcher.event_name`. So I suggest we do this in `main()`:

```python
    breaking_relation_id = None
    if dispatcher.event_name.endswith('_relation_broken'):
        breaking_relation_id = _parse_relation_id()
```